lib/Doctrine/DBAL/Driver/SQLSrv/SQLSrvConnection.php

```diff
@@ -124,7 +125,19 @@ public function exec($statement)
             throw SQLSrvException::fromSqlSrvErrors();
         }
 
-        return sqlsrv_rows_affected($stmt);
+        $affectedRowCount = sqlsrv_rows_affected($stmt);
+
+        $stmt = sqlsrv_query($this->conn, 'SELECT @@IDENTITY');
```

Comment:
Since this is an SQLSrv-only problem, this comment can be ignored.

IMO: their problem for relying on SQLSrv.